rxjava-modules/rxjava-observables/src/test/java/com/baeldung/rxjava/combine/ObservableMergeVsConcatUnitTest.java

```diff
@@ -0,0 +1,129 @@
+package com.baeldung.rxjava.combine;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Test;
+
+import rx.Observable;
+import rx.observers.TestSubscriber;
+
+public class ObservableMergeVsConcatUnitTest {
+
+    @Test
+    public void givenTwoSynchronousObservables_whenConcatenated_thenAllValuesEmittedInOrder() {
+
+        Observable<Integer> observable1 = Observable.just(1, 2, 3);
+        Observable<Integer> observable2 = Observable.just(4, 5, 6);
+
+        TestSubscriber<Integer> testSubscriberForConcat = new TestSubscriber<>();
+
+        Observable.concat(observable1, observable2)
+            .subscribe(testSubscriberForConcat);
+
+        testSubscriberForConcat.assertValues(1, 2, 3, 4, 5, 6);
+    }
+
+    @Test
+    public void givenTwoSynchronousObservables_whenMerged_thenAllValuesEmitted() {
+
+        Observable<Integer> observable1 = Observable.just(1, 2, 3);
+        Observable<Integer> observable2 = Observable.just(4, 5, 6);
+
+        TestSubscriber<Integer> testSubscriberForMerge = new TestSubscriber<>();
+
+        Observable.merge(observable1, observable2)
+            .subscribe(testSubscriberForMerge);
+
+        testSubscriberForMerge.assertValues(1, 2, 3, 4, 5, 6);
+    }
+
+    @Test
+    public void givenTwoAsynchronousObservables_whenConcatenated_thenAllValuesEmittedInOrder() {
+
+        Observable<Integer> observable1 = Observable.interval(100, TimeUnit.MILLISECONDS)
+            .map(i -> i.intValue() + 1)
+            .take(3);
+
+        Observable<Integer> observable2 = Observable.interval(10, TimeUnit.MILLISECONDS)
+            .map(i -> i.intValue() + 4)
+            .take(3);
+
+        TestSubscriber<Integer> testSubscriberForConcat = new TestSubscriber<>();
+
+        Observable.concat(observable1, observable2)
+            .subscribe(testSubscriberForConcat);
+
+        testSubscriberForConcat.awaitTerminalEvent();
+
+        testSubscriberForConcat.assertValues(1, 2, 3, 4, 5, 6);
+    }
+
+    @Test
+    public void givenTwoAsynchronousObservables_whenMerged_thenAllValuesEmittedInterleaved() {
+
+        Observable<Integer> observable1 = Observable.interval(100, TimeUnit.MILLISECONDS)
+            .map(i -> i.intValue() + 1)
+            .take(3);
+
+        Observable<Integer> observable2 = Observable.interval(10, TimeUnit.MILLISECONDS)
+            .map(i -> i.intValue() + 4)
+            .take(3);
+
+        TestSubscriber<Integer> testSubscriberForMerge = new TestSubscriber<>();
+
+        Observable.merge(observable1, observable2)
+            .subscribe(testSubscriberForMerge);
+
+        testSubscriberForMerge.awaitTerminalEvent();
+
+        testSubscriberForMerge.assertValues(4, 5, 6, 1, 2, 3);
+    }
```

Comment:
Is this a deterministic ordering?

Interleaved would be more like 1, 4, 2, 5, 3, 6

This just seems to have put the observable2 values before the observable1 ones

Do you understand why the order is 4, 5, 6, 1, 2, 3?

Ah I think I got it, observable1 is 100ms and observable2 is 10ms interval. Hmm okay, I think this might require a bit more explanation in the article to avoid confusion, because it doesn't actually demonstrate an interleaving.

Thoughts?

Reply:
As per your suggestion, we have now changed the number of emissions to depict a clear interleaving. I think it explains the concept more clearly now. Thanks!